### ISSUE TITLE
update to last xcode and flutter/dart versione

### ISSUE THREAD

**Issue by GabriAlbarosa** [flutter/flutter#155400]
### Steps to reproduce

build with versione 6.2.5 or latest

### Expected results

cannot istall app

### Actual results

Swift Compiler Error (Xcode): Type 'UIApplication' does not conform to protocol 'Launcher'
/Users/gabrielalbarosa/.pub-cache/hosted/pub.dev/url_launcher_ios-6.2.5/ios/Classes/Launcher.swift:21:0

Could not build the application for the simulator.
Error launching application on iPhone 16 Pro Max.

### Code sample

not working

### Screenshots or Video

not working

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel master, 3.26.0-1.0.pre.161, on macOS 15.0 24A335 darwin-arm64, locale it-IT)
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    ! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses
[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] VS Code (version 1.93.1)
[✓] Connected device (6 available)
[✓] Network resources

! Doctor found issues in 1 category.
```

</details>

**Comment by danagbemava-nc** [comment]
This has been fixed in url_launcher_ios: 6.3.1. See https://github.com/flutter/flutter/issues/155306 and try https://github.com/flutter/flutter/issues/155306#issuecomment-2357294055 to see if it helps. I just tested the url_launcher plugin on stable and master with the latest Xcode and it works as expected.

**Comment by stuartmorgan** [comment]
> /Users/gabrielalbarosa/.pub-cache/hosted/pub.dev/url_launcher_ios-6.2.5/ios/Classes/Launcher.swift:21:0

6.2.5 is too old to have the fix; closing as a duplicate.
